protocol/msgpack/unaggregated_iterator.go

```diff
@@ -166,12 +173,28 @@ func (it *unaggregatedIterator) decodeBatchTimer() {
 	}
 	it.metric.Type = unaggregated.BatchTimerType
 	it.metric.ID = it.decodeID()
-	numValues := it.decodeArrayLen()
-	values := it.floatsPool.Get(numValues)
+	var (
+		timerValues []float64
+		poolAlloc   = false
+		numValues   = it.decodeArrayLen()
+	)
+	if cap(it.timerValues) >= numValues {
+		it.timerValues = it.timerValues[:0]
+		timerValues = it.timerValues
+	} else if numValues <= it.largeFloatsSize {
+		it.timerValues = make([]float64, 0, numValues)
```

Comment:
Gotcha. Yea this sounds good to me. I was just thinking that if we're pooling aggregated iterators in the aggregator then after its been running for a while the `timerValues` will probably all reach `largeFloatsSize ` in which case it might be better to set the capacity to that when we construct it.